src/commander.c

```diff
@@ -448,6 +429,30 @@ static void commander_process_backup(yajl_val json_node)
         return;
     }
 
+
+    if (wallet_is_locked()) {
```

Comment:
the diff is hard to read. All I did was to move the those lines:

```C
if (strlens(value)) {
        if (STREQ(value, attr_str(ATTR_list))) {
            sd_list(CMD_backup);
            return;
        }

        commander_fill_report(cmd_str(CMD_backup), NULL, DBB_ERR_IO_INVALID_CMD);
        return;
    }

    if (strlens(source_y)) {
        snprintf(source, sizeof(source), "%s", source_y);
    } else {
        // Defaults
        if (check) {
            snprintf(source, sizeof(source), "%s", attr_str(ATTR_HWW));// Backward compatible
        } else {
            snprintf(source, sizeof(source), "%s", attr_str(ATTR_all));
        }
    }

    if (check) {
        // Verify existing backup
        if (STREQ(source, attr_str(ATTR_all))) {
            commander_fill_report(cmd_str(CMD_backup), NULL, DBB_ERR_IO_INVALID_CMD);
            return;
        }
        commander_process_backup_check(key, check, source);
        return;
    }
```

from below this line to above this line (pass even if device is locked (2fa is enabled)).